doc/sphinx/language/core/basic.rst

```diff
@@ -146,6 +148,29 @@ Numbers
   The :cmd:`Number Notation` mechanism offers the user
   a way to define custom parsers and printers for :n:`@number`.
 
+  By default, numbers are interpreted as :n:`nat`\s, which is a unary
+  representation.  For example, :n:`3` is represented as `S (S (S O))`.  While
+  this is a convenient representation for doing proofs, computing with large
+  :n:`nat`\s can lead to stack overflows or running out of memory.  You can
+  explicitly specify a different interpretation to avoid this problem.  For
+  example, :n:`1000000%N` efficiently represents that natural number.  See
+  :ref:`Scopes` and :n:`@term_scope`.  You can also change the default
+  interpretation using the :cmd:`Number Notation` command.
+
+   .. example:: Stack overflow with :n:`nat`
+
+      .. coqtop:: all reset
+
+         Fail Compute 100000 + 100000.  (* gives a stack overflow (not shown) *)
```

Comment:
```suggestion
         Fail Eval compute in 100000 + 100000.  (* gives a stack overflow (not shown) *)
```
(equivalent but `Compute` probably only remains for historical reasons from an old time there was only one evaluation mechanism in Coq)

Reply:
I think `Compute` is widely used and it's simpler.  Software Foundations, for example, uses `Compute`.